presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetPageSourceFactoryUtils.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.hive.parquet;
+
+import com.facebook.presto.hive.HiveFileSplit;
+import com.facebook.presto.parquet.ParquetCorruptionException;
+import com.facebook.presto.spi.PrestoException;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.security.AccessControlException;
+import org.apache.parquet.crypto.HiddenColumnException;
+
+import java.io.FileNotFoundException;
+
+import static com.facebook.presto.hive.HiveErrorCode.HIVE_BAD_DATA;
+import static com.facebook.presto.hive.HiveErrorCode.HIVE_CANNOT_OPEN_SPLIT;
+import static com.facebook.presto.hive.HiveErrorCode.HIVE_MISSING_DATA;
+import static com.facebook.presto.spi.StandardErrorCode.PERMISSION_DENIED;
+import static com.google.common.base.Strings.nullToEmpty;
+import static java.lang.String.format;
+
+public class ParquetPageSourceFactoryUtils
+{
+    private ParquetPageSourceFactoryUtils() {}
+
+    public static PrestoException mapToPrestoException(Exception e, Path path, HiveFileSplit fileSplit)
```

Comment:
this was copied wholesale from the code that had been in ParquetPageSourceFactory. No changes here.

Reply:
We can probably generalize common exceptions and keep them in a all purpose PageSourceFactoryUtils